taipy/gui_core/_context.py

```diff
@@ -1217,18 +1219,19 @@ def on_dag_select(self, state: State, id: str, payload: t.Dict[str, str]):
 
     def on_file_action(self, state: State, id: str, payload: t.Dict[str, t.Any]):
         args = t.cast(list, payload.get("args"))
+        if args is None or not isinstance(args, list) or len(args) < 1 or not isinstance(args[0], dict):
+            return
         act_payload = t.cast(t.Dict[str, str], args[0])
         dn_id = t.cast(DataNodeId, act_payload.get("id"))
         error_id = act_payload.get("error_id", "")
         if reason := is_readable(dn_id):
             try:
                 dn = t.cast(_FileDataNodeMixin, core_get(dn_id))
                 if act_payload.get("action") == "export":
-                    path = dn._get_downloadable_path()
-                    if path:
+                    if reason := dn.is_downloadable():
+                        path = dn._get_downloadable_path()
                         self.gui._download(Path(path), dn_id)
```

Comment:
no need to check path if is_downloadable ?

Reply:
Absolutely. That's what `is_downloadable` does. It builds the Path, checks if it exists, and if it is a file.